xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
Sorry I don't follow. I would have expected a scenario that shows how Kodi fails in any of its functions.

I agree that there is a race but this has no influence on Kodi's behavior.

Note that m_CriticalSection is private so your example above won't work